backend/handler/handler.go

```diff
@@ -125,6 +127,37 @@ func (h *handler) getTables(rw http.ResponseWriter, req *http.Request) {
 	writeHTTPResult(rw, tables)
 }
 
+// getVariables is a handle to fetch variable values
+func (h *handler) getVariables(rw http.ResponseWriter, req *http.Request) {
+	backend.Logger.Debug("Process 'variables' request")
+
+	pluginCtx := httpadapter.PluginConfigFromContext(req.Context())
+	p, err := h.getPluginInstance(req.Context(), pluginCtx)
+	if err != nil {
+		backend.Logger.Error("Failed to get plugin instance", "Message", err)
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	query := req.URL.Query().Get("query")
+
+	variables, err := p.GetVariables(req.Context(), query)
+	if err != nil {
+		backend.Logger.Error("Failed to get variables", "Message", err)
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	serialisedVars, err := json.Marshal(variables)
```

Comment:
I'm really unsure about this idea but found no better way. Grafana's getResource method supports only `[]string`, but we have to support values AND labels. Feel free to suggest something better.

Reply:
It looks like `getResource` is not the best way to do that. What do you think about doing that the way `queryMetricData` does? It would look more straightforward - handler for variable query and array of custom objects to return data to frontend, where it would be passed to array of `MetricFindValue`.

Reply:
That might be an option. Moreover, that might be the right way, lol. Not so happy about the refactoring required though ahahaha